pipenv/utils/resolver.py

```diff
@@ -308,7 +307,7 @@ def pip_options(self):
         )
         return pip_options
 
-    @property
+    @cached_property
```

Comment:
I did some performance profiling and this was a huge win -- there was also an issue with too many sessions and closing the sockets, which I experienced myself recently and this absolutely fixed that.   Couldn't locate the issue report at this time.